src/dat.rs

```diff
@@ -0,0 +1,231 @@
+            {
+                (1, 0) => {
+                    let content = decrypt(&bytes, pwd)?;
+                    let mut buff = &content[..];
+
+                    let seed = store::Seed::from_reader(&mut buff)
+                        .map_err(|_| Error::WalletFileCorrupted)?;
+
+                    Ok((seed, 1))
+                }
+                (2, 0) => {
+                    let content = decrypt(&bytes, pwd)?;
+                    let mut buff = &content[..];
+
+                    // extract seed
+                    let seed = store::Seed::from_reader(&mut buff)
+                        .map_err(|_| Error::WalletFileCorrupted)?;
+
+                    // extract addresses count
+                    Ok((seed, buff[0]))
+                }
+                _ => Err(Error::UnknownFileVersion(major, minor)),
+            };
+
+            result
+        }
+        DatFileVersion::RuskBinaryFileFormat(_) => {
+            let rest = bytes.get(12..(12 + 96));
+            if let Some(rest) = rest {
+                let content = decrypt(rest, pwd)?;
+
+                if let Some(seed_buff) = content.get(0..65) {
+                    let seed = store::Seed::from_reader(&mut &seed_buff[0..64])
+                        .map_err(|_| Error::WalletFileCorrupted)?;
+
+                    let count = &seed_buff[64..65];
+
+                    Ok((seed, count[0]))
+                } else {
+                    Err(Error::WalletFileCorrupted)
+                }
+            } else {
+                Err(Error::WalletFileCorrupted)
+            }
+        }
+    }
+}
+
+/// From the first 12 bytes of the file (header), we check version
+///
+/// https://github.com/dusk-network/rusk/wiki/Binary-File-Format/#header
+pub(crate) fn check_version(
+    bytes: Option<&[u8]>,
+) -> Result<DatFileVersion, Error> {
+    match bytes {
+        Some(bytes) => {
+            let header_bytes: [u8; 4] = bytes[0..4]
+                .try_into()
+                .map_err(|_| Error::WalletFileCorrupted)?;
+
+            let magic = u32::from_le_bytes(header_bytes) & 0x00ffffff;
+
+            if magic == OLD_MAGIC {
+                // check for version information
+                let (major, minor) = (bytes[3], bytes[4]);
+
+                Ok(DatFileVersion::OldWalletCli((major, minor, 0, 0, false)))
+            } else {
+                let header_bytes = bytes[0..8]
+                    .try_into()
+                    .map_err(|_| Error::WalletFileCorrupted)?;
+
+                let number = u64::from_be_bytes(header_bytes);
+
+                let magic_num = (number & 0xFFFFFF00000000) >> 32;
+
+                if (magic_num as u32) != MAGIC {
+                    return Ok(DatFileVersion::Legacy);
+                }
+
+                let file_type = (number & 0x000000FFFF0000) >> 16;
+                let reserved = number & 0x0000000000FFFF;
+
+                if file_type != 0x200 {
```

Comment:
Can you add the file_type as a const like `MAGIC`

nit: pls specify as 0x0200 instead of 0x200